src/terminal/input/terminalInput.cpp

```diff
@@ -169,111 +185,143 @@ const TerminalInput::_TermKeyMap TerminalInput::s_rgModifierKeyMapping[]{
 // These sequences are not later updated to encode the modifier state in the
 //      sequence itself, they are just weird exceptional cases to the general
 //      rules above.
-const TerminalInput::_TermKeyMap TerminalInput::s_rgSimpleModifedKeyMapping[]{
-    // HEY YOU. UPDATE THE MAX LENGTH DEF WHEN YOU MAKE CHANGES HERE.
-    { VK_BACK, CTRL_PRESSED, L"\x8" },
-    { VK_BACK, ALT_PRESSED, L"\x1b\x7f" },
-    { VK_BACK, CTRL_PRESSED | ALT_PRESSED, L"\x1b\x8" },
-    { VK_TAB, CTRL_PRESSED, L"\t" },
-    { VK_TAB, SHIFT_PRESSED, L"\x1b[Z" },
-    { VK_DIVIDE, CTRL_PRESSED, L"\x1F" },
+static constexpr std::array<TermKeyMap, 6> s_simpleModifedKeyMapping{
+    TermKeyMap{ VK_BACK, CTRL_PRESSED, L"\x8" },
+    TermKeyMap{ VK_BACK, ALT_PRESSED, L"\x1b\x7f" },
+    TermKeyMap{ VK_BACK, CTRL_PRESSED | ALT_PRESSED, L"\x1b\x8" },
+    TermKeyMap{ VK_TAB, CTRL_PRESSED, L"\t" },
+    TermKeyMap{ VK_TAB, SHIFT_PRESSED, L"\x1b[Z" },
+    TermKeyMap{ VK_DIVIDE, CTRL_PRESSED, L"\x1F" },
     // These two are not implemented here, because they are system keys.
-    // { VK_TAB, ALT_PRESSED, L""}, This is the Windows system shortcut for switching windows.
-    // { VK_ESCAPE, ALT_PRESSED, L""}, This is another Windows system shortcut for switching windows.
+    // TermKeyMap{ VK_TAB, ALT_PRESSED, L""}, This is the Windows system shortcut for switching windows.
+    // TermKeyMap{ VK_ESCAPE, ALT_PRESSED, L""}, This is another Windows system shortcut for switching windows.
 };
 
 const wchar_t* const CTRL_SLASH_SEQUENCE = L"\x1f";
 
-// Do NOT include the null terminator in the count.
-const size_t TerminalInput::_TermKeyMap::s_cchMaxSequenceLength = 7; // UPDATE THIS DEF WHEN THE LONGEST MAPPED STRING CHANGES
-
-const size_t TerminalInput::s_cCursorKeysNormalMapping = ARRAYSIZE(s_rgCursorKeysNormalMapping);
-const size_t TerminalInput::s_cCursorKeysApplicationMapping = ARRAYSIZE(s_rgCursorKeysApplicationMapping);
-const size_t TerminalInput::s_cKeypadNumericMapping = ARRAYSIZE(s_rgKeypadNumericMapping);
-const size_t TerminalInput::s_cKeypadApplicationMapping = ARRAYSIZE(s_rgKeypadApplicationMapping);
-const size_t TerminalInput::s_cModifierKeyMapping = ARRAYSIZE(s_rgModifierKeyMapping);
-const size_t TerminalInput::s_cSimpleModifedKeyMapping = ARRAYSIZE(s_rgSimpleModifedKeyMapping);
-
-void TerminalInput::ChangeKeypadMode(const bool fApplicationMode)
+void TerminalInput::ChangeKeypadMode(const bool applicationMode) noexcept
 {
-    _fKeypadApplicationMode = fApplicationMode;
+    _keypadApplicationMode = applicationMode;
 }
 
-void TerminalInput::ChangeCursorKeysMode(const bool fApplicationMode)
+void TerminalInput::ChangeCursorKeysMode(const bool applicationMode) noexcept
 {
-    _fCursorApplicationMode = fApplicationMode;
+    _cursorApplicationMode = applicationMode;
 }
 
-const size_t TerminalInput::GetKeyMappingLength(const KeyEvent& keyEvent) const
+const std::basic_string_view<TermKeyMap> _getKeyMapping(const KeyEvent& keyEvent,

```

Comment:
the functions moved into this file can be marked `static` so they do not leak into global scope.

Reply:
OK.